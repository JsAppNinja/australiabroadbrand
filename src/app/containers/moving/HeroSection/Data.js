import MovingHeroLeft from '../../../assets/moving-hero-left.png';
import MovingHeroRight from '../../../assets/moving-hero-right.png';

export default {
  herotitle: 'Hassle free nbn™ moving service',
  heroDescription:
    'If you’re moving into a new place and need to connect to the nbn™, Australia Broadband provides an easy to switch & easy to connect service with no lock in contracts and no worries.',
  buttonContent: 'Check address',
  hero_left: MovingHeroLeft,
  hero_right: MovingHeroRight,
};
