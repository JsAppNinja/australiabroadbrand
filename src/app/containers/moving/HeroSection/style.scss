.moving-hero {
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-weight: 500;
  padding: 108px 130px 105px 105px;
  background-color: #f6f9ff;

  .moving-hero-action {
    display: flex;
    justify-content: center;
    flex-direction: column;
    text-align: left;
    width: 505px;

    .moving-hero-action__title {
      font-family: Baloo;
      font-size: 28px;
      line-height: 1;
      color: #4d565a;
    }

    .moving-hero-action__desc {
      font-family: Roboto;
      font-size: 14px;
      line-height: 24px;
      letter-spacing: 0;
      padding-top: 17px;
      color: #313541;
      font-weight: bold;
    }

    .moving-hero-action__button {
      padding-top: 42px;
      cursor: pointer;
    }
  }
  .moving-hero-visual {
    display: flex;
    align-items: center;
    img {
      &:first-child {
        height: 165px;
        width: 340px;
        margin-right: 20px;
      }
      &:nth-child(2) {
        max-height: 265px;
        max-width: 255px;
      }
    }
  }
}
