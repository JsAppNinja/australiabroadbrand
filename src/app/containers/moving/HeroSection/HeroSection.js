import React, { Component } from 'react';
import CheckButton from '../../../components/CheckButton';

import './style.scss';
import Data from './Data';

class HeroSection extends Component {
  render() {
    const customStyle = {
      width: '230px',
      height: '45px',
      backgroundColor: '#67cebb',
      color: '#ffffff',
      borderRadius: '22.5px',
      fontSize: '18px',
    };

    return (
      <div className="moving-hero">
        <div className="moving-hero-action">
          <div className="moving-hero-action__title">{Data.herotitle}</div>
          <div className="moving-hero-action__desc">{Data.heroDescription}</div>
          <div className="moving-hero-action__button">
            <CheckButton
              CustomStyle={customStyle}
              Content={Data.buttonContent}
            />
          </div>
        </div>
        <div className="moving-hero-visual">
          <img src={Data.hero_left} alt="" />
          <img src={Data.hero_right} alt="" />
        </div>
      </div>
    );
  }
}

export default HeroSection;
