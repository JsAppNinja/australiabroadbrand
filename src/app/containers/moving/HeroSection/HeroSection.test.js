import React from 'react';
import { shallow, mount, render } from '../../../../enzyme.js';

import HeroSection from './HeroSection.js';

describe('HeroSection tests', () => {
  const tree = shallow(<HeroSection />);

  it('should render Hero section', () => {
    expect(tree).toMatchSnapshot();
  });
  it('should be selectable by class "moving-hero-wrapper"', function() {
    expect(shallow(<HeroSection />).is('.moving-hero-wrapper')).toBe(true);
  });

  it('should mount in a full DOM', function() {
    expect(mount(<HeroSection />).find('.moving-hero').length).toBe(1);
  });
});
