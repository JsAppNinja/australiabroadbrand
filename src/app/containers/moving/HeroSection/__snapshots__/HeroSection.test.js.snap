// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroSection tests should render Hero section 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeroSection />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="moving-hero"
      >
        <div
          className="moving-hero-action"
        >
          <div
            className="moving-hero-action__title"
          >
            Hassle free nbn™ moving service
          </div>
          <div
            className="moving-hero-action__desc"
          >
            If you’re moving into a new place and need to connect to the nbn™, Australia Broadband provides an easy to switch & easy to connect service with no lock in contracts and no worries.
          </div>
          <div
            className="moving-hero-action__button"
          >
            <CheckButton
              Content="Check address"
              CustomStyle={
                Object {
                  "backgroundColor": "#67cebb",
                  "borderRadius": "22.5px",
                  "color": "#ffffff",
                  "fontSize": "18px",
                  "height": "45px",
                  "width": "230px",
                }
              }
            />
          </div>
        </div>
        <div
          className="moving-hero-visual"
        >
          <img
            alt=""
            src="moving-hero-left.png"
          />
          <img
            alt=""
            src="moving-hero-right.png"
          />
        </div>
      </div>,
      "className": "moving-hero-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="moving-hero-action"
          >
            <div
              className="moving-hero-action__title"
            >
              Hassle free nbn™ moving service
            </div>
            <div
              className="moving-hero-action__desc"
            >
              If you’re moving into a new place and need to connect to the nbn™, Australia Broadband provides an easy to switch & easy to connect service with no lock in contracts and no worries.
            </div>
            <div
              className="moving-hero-action__button"
            >
              <CheckButton
                Content="Check address"
                CustomStyle={
                  Object {
                    "backgroundColor": "#67cebb",
                    "borderRadius": "22.5px",
                    "color": "#ffffff",
                    "fontSize": "18px",
                    "height": "45px",
                    "width": "230px",
                  }
                }
              />
            </div>
          </div>,
          <div
            className="moving-hero-visual"
          >
            <img
              alt=""
              src="moving-hero-left.png"
            />
            <img
              alt=""
              src="moving-hero-right.png"
            />
          </div>,
        ],
        "className": "moving-hero",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="moving-hero-action__title"
              >
                Hassle free nbn™ moving service
              </div>,
              <div
                className="moving-hero-action__desc"
              >
                If you’re moving into a new place and need to connect to the nbn™, Australia Broadband provides an easy to switch & easy to connect service with no lock in contracts and no worries.
              </div>,
              <div
                className="moving-hero-action__button"
              >
                <CheckButton
                  Content="Check address"
                  CustomStyle={
                    Object {
                      "backgroundColor": "#67cebb",
                      "borderRadius": "22.5px",
                      "color": "#ffffff",
                      "fontSize": "18px",
                      "height": "45px",
                      "width": "230px",
                    }
                  }
                />
              </div>,
            ],
            "className": "moving-hero-action",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Hassle free nbn™ moving service",
                "className": "moving-hero-action__title",
              },
              "ref": null,
              "rendered": "Hassle free nbn™ moving service",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "If you’re moving into a new place and need to connect to the nbn™, Australia Broadband provides an easy to switch & easy to connect service with no lock in contracts and no worries.",
                "className": "moving-hero-action__desc",
              },
              "ref": null,
              "rendered": "If you’re moving into a new place and need to connect to the nbn™, Australia Broadband provides an easy to switch & easy to connect service with no lock in contracts and no worries.",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <CheckButton
                  Content="Check address"
                  CustomStyle={
                    Object {
                      "backgroundColor": "#67cebb",
                      "borderRadius": "22.5px",
                      "color": "#ffffff",
                      "fontSize": "18px",
                      "height": "45px",
                      "width": "230px",
                    }
                  }
                />,
                "className": "moving-hero-action__button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "Content": "Check address",
                  "CustomStyle": Object {
                    "backgroundColor": "#67cebb",
                    "borderRadius": "22.5px",
                    "color": "#ffffff",
                    "fontSize": "18px",
                    "height": "45px",
                    "width": "230px",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt=""
                src="moving-hero-left.png"
              />,
              <img
                alt=""
                src="moving-hero-right.png"
              />,
            ],
            "className": "moving-hero-visual",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "src": "moving-hero-left.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "src": "moving-hero-right.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="moving-hero"
        >
          <div
            className="moving-hero-action"
          >
            <div
              className="moving-hero-action__title"
            >
              Hassle free nbn™ moving service
            </div>
            <div
              className="moving-hero-action__desc"
            >
              If you’re moving into a new place and need to connect to the nbn™, Australia Broadband provides an easy to switch & easy to connect service with no lock in contracts and no worries.
            </div>
            <div
              className="moving-hero-action__button"
            >
              <CheckButton
                Content="Check address"
                CustomStyle={
                  Object {
                    "backgroundColor": "#67cebb",
                    "borderRadius": "22.5px",
                    "color": "#ffffff",
                    "fontSize": "18px",
                    "height": "45px",
                    "width": "230px",
                  }
                }
              />
            </div>
          </div>
          <div
            className="moving-hero-visual"
          >
            <img
              alt=""
              src="moving-hero-left.png"
            />
            <img
              alt=""
              src="moving-hero-right.png"
            />
          </div>
        </div>,
        "className": "moving-hero-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="moving-hero-action"
            >
              <div
                className="moving-hero-action__title"
              >
                Hassle free nbn™ moving service
              </div>
              <div
                className="moving-hero-action__desc"
              >
                If you’re moving into a new place and need to connect to the nbn™, Australia Broadband provides an easy to switch & easy to connect service with no lock in contracts and no worries.
              </div>
              <div
                className="moving-hero-action__button"
              >
                <CheckButton
                  Content="Check address"
                  CustomStyle={
                    Object {
                      "backgroundColor": "#67cebb",
                      "borderRadius": "22.5px",
                      "color": "#ffffff",
                      "fontSize": "18px",
                      "height": "45px",
                      "width": "230px",
                    }
                  }
                />
              </div>
            </div>,
            <div
              className="moving-hero-visual"
            >
              <img
                alt=""
                src="moving-hero-left.png"
              />
              <img
                alt=""
                src="moving-hero-right.png"
              />
            </div>,
          ],
          "className": "moving-hero",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="moving-hero-action__title"
                >
                  Hassle free nbn™ moving service
                </div>,
                <div
                  className="moving-hero-action__desc"
                >
                  If you’re moving into a new place and need to connect to the nbn™, Australia Broadband provides an easy to switch & easy to connect service with no lock in contracts and no worries.
                </div>,
                <div
                  className="moving-hero-action__button"
                >
                  <CheckButton
                    Content="Check address"
                    CustomStyle={
                      Object {
                        "backgroundColor": "#67cebb",
                        "borderRadius": "22.5px",
                        "color": "#ffffff",
                        "fontSize": "18px",
                        "height": "45px",
                        "width": "230px",
                      }
                    }
                  />
                </div>,
              ],
              "className": "moving-hero-action",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Hassle free nbn™ moving service",
                  "className": "moving-hero-action__title",
                },
                "ref": null,
                "rendered": "Hassle free nbn™ moving service",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "If you’re moving into a new place and need to connect to the nbn™, Australia Broadband provides an easy to switch & easy to connect service with no lock in contracts and no worries.",
                  "className": "moving-hero-action__desc",
                },
                "ref": null,
                "rendered": "If you’re moving into a new place and need to connect to the nbn™, Australia Broadband provides an easy to switch & easy to connect service with no lock in contracts and no worries.",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <CheckButton
                    Content="Check address"
                    CustomStyle={
                      Object {
                        "backgroundColor": "#67cebb",
                        "borderRadius": "22.5px",
                        "color": "#ffffff",
                        "fontSize": "18px",
                        "height": "45px",
                        "width": "230px",
                      }
                    }
                  />,
                  "className": "moving-hero-action__button",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "Content": "Check address",
                    "CustomStyle": Object {
                      "backgroundColor": "#67cebb",
                      "borderRadius": "22.5px",
                      "color": "#ffffff",
                      "fontSize": "18px",
                      "height": "45px",
                      "width": "230px",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt=""
                  src="moving-hero-left.png"
                />,
                <img
                  alt=""
                  src="moving-hero-right.png"
                />,
              ],
              "className": "moving-hero-visual",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "src": "moving-hero-left.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "src": "moving-hero-right.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
